exercise/submission_models.py

```diff
@@ -613,6 +625,22 @@ def get_url_kwargs(self):
     def get_inspect_url(self):
         return self.get_url("submission-inspect")
 
+    def mark_pending(self):
+        grading_host = urlparse(self.exercise.service_url).netloc
+        if grading_host in settings.SUBMISSION_RETRY_SERVICES:
+            pending, created = PendingSubmission.objects.get_or_create(submission=self)
+            if not created:
+                pending.num_retries = F('num_retries') + 1
+            pending.submission_time = timezone.now()
+            pending.save()
+
+    def clear_pending(self):
+        try:
+            pending = PendingSubmission.objects.get(submission=self)
+            pending.delete()
+        except PendingSubmission.DoesNotExist:
+            logger.warn(f"Submission {self} completed, but was not on PendingSubmissions table.")
```

Comment:
Isn't this going to print warnings for all submissions in exercise services that are not included in `settings.SUBMISSION_RETRY_SERVICES`?

Reply:
Oops, indeed. Perhaps it is best just to silently ignore this exception. Changed this.